.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+name: CI
+
+on: push
+
+jobs:
+
+  standard:
+
+    strategy:
+      fail-fast: false
+      matrix:
+        xsimd: [-DXTENSOR_USE_XSIMD=ON, -DXTENSOR_USE_XSIMD=OFF]
+        compiler: [g++, clang++, cl]
+        runs-on: [ubuntu-latest, macos-latest, windows-latest]
+        exclude:
+        - runs-on: macos-latest
+          compiler: cl
+        - runs-on: macos-latest
+          compiler: g++
+        - runs-on: ubuntu-latest
+          compiler: cl
```

Comment:
This is something we should discuss. E.g. @JohanMabille  should comment. 

I think we could run a bit less aggressively. This PR implies 3 x 3 x 2 = 18 - 3 x 2 = 12 runs if I'm not mistaken. That is not accounting for the options that have not yet be included :

https://github.com/xtensor-stack/xtensor/blob/master/.azure-pipelines/azure-pipelines-linux-gcc.yml

Although this PR will thus be consistent with the current 21 runs, that are each quite long, this is not in line with my ecological convictions, and I feel also unneeded. Can't we enforce only 3-6 runs on PRs and maybe some more on changes to the master? I presume that the latter could be hand-triggered on PRs if addressing specific compiler bugs.

Reply:
All the options (with or without xsimd, column_major layour, disable_exceptions) should be tested; but we can probably reduce the number of compilers we test, this was relevant when the project was started because C++14 was quite recent and compilers have some bugs. However, I recommend not to test the latest versions of the compilers only.

Maybe a more explicit matrix (like [this one](https://github.com/xtensor-stack/xsimd/blob/master/.github/workflows/linux.yml#L16) for instance) would be better in our case.